### PR TITLE
[QOL] TPD Nearest Hour

### PR DESCRIPTION
**Pull request by brianleect**
Default initializes TPD start time to nearest hour. 

Should fix 0% issue at the same time as well since nearest hour would act as a buffer. 
(Although not the original intention, it's interesting how it happens to fix the issue)

### CODE REVIEW

**Comment by brianleect** [comment]
Will try running this version overnight and if it works fine will merge it in the morning.

**Comment by brianleect** [comment]
image

Logs also show how long before the start time is considered for the TPD

**Comment by brianleect** [comment]
Will probably need to look at the modification related to the docker files to add this new parameter.

File: entrypoint.sh [hunk 1]
```diff
@@ -72,6 +72,9 @@ process_config() {
     if [[ -n $CHECK_NEW_LISTING_ENABLED ]]; then
         sed -i "s/checkNewListingEnabled.*/checkNewListingEnabled: ${CHECK_NEW_LISTING_ENABLED}/" config.yml
     fi
+    if [[ -n $TPD_NEAREST_HOUR ]]; then
+        sed -i "s/tpdNearestHour.*/tpdNearestHour: ${TPD_NEAREST_HOUR}/" config.yml
+    fi
```
Comment (brianleect) [inline entrypoint.sh:77]: @patbaumgartner Is this the proper method to add the parameter to the entrypoint file for docker?

I followed the naming convention for the other parameters above, but I'm not sure if it's correct and whether changes in other areas will be needed.

**Comment by brianleect** [comment]
image

Based on testing so far, 0% for first TPD is fixed as expected.

Once there's no issue with the docker side, should be able to merge this PR.

**Comment by patbaumgartner** [comment]
Why don't you call the setting `topReportNearestHour: True` it would be way more consistent with the current code.

You should also add the search & replace in the `entrypoint.sh`  which is straight forward especially you can take it over from DEBUG. Adding it to the `docker-compose.yml` is not needed.

**Pull request merged**